docs/torchbench.md

```diff
@@ -0,0 +1,203 @@
+number of FX graphs compiled successfully by AOTAutograd, inductor related information,
+etc.
+
+**Profiler Metrics for CPU and CUDA:** the flag `--profile-cuda-cpu` collects CPU and CUDA
+profiling information about the execution. Specifically, it collects 4 of them:
+
+1. `total_cpu_time_s`: total CPU time take for running all iterations
+2. `per_iter_cpu_time_s`: average CPU time taken on each iteration
+3. `total_cuda_time_s`: total CUDA time take for running all iterations
+4. `per_iter_cuda_time_s`: average CPU time taken on each iteration
+
+**Profiler Metrics for CPU and CUDA for unlowered operations:** the flag
+`--profile-cuda-cpu-individual-ops` collects profiling information for operations that
+went through the fallback path, i.e. those for which PyTorch/XLA does not have a
+lowering. For each of those operations, the scripts collect 5 new metrics:
+
+1. `self_cpu_time_s`: CPU time spent in that operation
+2. `total_cpu_time_s`: CPU time spent running that function, including other dependent
+   operations
+3. `self_cuda_time_s`: CUDA time spent in that operation
+4. `total_cuda_time_s`: CUDA time spent running that function, including other dependent
+   operations
+5. `num_of_calls`: number of times that operation was called
+
+### Dumping Benchmark-specific Data
+
+Besides the collected metrics, the benchmarking scripts have the option for dumping
+information specific to each benchmark. These are stored in a newly created directory for
+each benchmark, under the output directory. The descriptions below assume all paths are
+relative to the benchmark-specific directory.
+
+There is information that is collected for each repetition (controlled by `--repeat`
+parameter) of a benchmark. Those are stored in separate directories, one for each
+repetition. For better clarity, we shall refer to those flags by marking them with the
+_"repeat-specific"_ tag.
+
+**HLO Modules:** the flag `--dump-hlo` sets the environment variable `XLA_FLAGS`, dumping
+XLA compiled HLO modules into the _"hlo"_ directory.
+
+**Dynamo Counters:** (repeat-specific) the flag `--dump-dynamo-counters` is similar to the
+already mentioned `--collect-dynamo-counters` flag. However, instead of collecting them
+and storing in a JSON line inside _results.jsonl_, setting this flag will dump the dynamo
+counter dictionary into a new file.
+
+**PyTorch Profiles and Trace:** (repeat-specific) the flag `--dump-pytorch-profiles`
+creates 2 new files. For more detailed information, check out [this post][9] explaining
+PyTorch profiling:
+
+1. Chrome trace: same as calling `profiler.export_chrome_trace`
+2. Profile: same as calling `profiler.key_averages().table(...)`, a summarized table with
+   the timing averages
+
+**PyTorch/XLA Metrics:** (repeat-specific) the flag `--dump-pytorch-xla-metrics` creates a
+new file, dumping PyTorch/XLA metrics, such as graph compiling and execution information.
+
```

Comment:
Can we have a section on Nighly CI runs - TPU and GPU

Reply:
Sure. But I don't really have any information on those nightly CI runs. What should it contain?

Reply:
@zpcore can you please help Yukio with this section?

Reply:
additional: ideally, we also want to run verifier as part of our tests